PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/graph/nodes/StraightSequenceBubble.java

```diff
@@ -37,12 +38,23 @@ public GraphNode copy() {
   public GraphNode copyAll() {
     return new StraightSequenceBubble(this, aligner);
   }
+  
+  @Override
+  public Collection<Integer> getGenomes() {
+    HashSet<Integer> genomeSet = new HashSet<>();
+    for (GraphNode nestedNode : getChildren()) {
+      genomeSet.addAll(nestedNode.getGenomes());
+    }
+    for (GraphNode inEdge : getInEdges()) {
+      genomeSet.addAll(inEdge.getGenomes());
+    }
```

Comment:
It is needed because the way the straight sequence is implemented now (without the first and last node) some genomes can be lost in the sequence bubble itself when its nested node is an indel. Because in this case it's still perceived as a straight sequence, but in reality it's not because there is an edge directly from the start to end node. So I do need to include the genomes of the node before for the filtering algorithm to work correctly. 

I think I could actually just add the genomes of the startnode, since all genomes that pass through there will also be in the rest of the sequence.